server/utils.js

```diff
@@ -29,8 +29,17 @@ var generateId = function(collection) {
 	return collection.findOne({guid: id}).then((data) => data ? generateId(db) : id);
 }
 
+let isLessThan = (dest, thenable) => (src) => {
```

Comment:
I'd disagree, if you use a replacement like the one I put later.  It was unclear to me what `isLessThan` meant on first read, whereas `version < 1 ? upgradeToOne() : version` would be immediately clear.

If you want to keep this, I'd at least move it out of `utils.js`, since it'd probably be clearer what its intended use is if it's with the upgrade-related code.